test/TrustedExecution.t.sol

```diff
@@ -0,0 +1,451 @@
+            address(characterNFTs),
+            PitcherTokenID,
+            address(otherCharacterNFTs),
+            BatterTokenID
+        );
+        emit AtBatProgress(
+            initialNumAtBats + 1,
+            AtBatOutcome.InProgress,
+            1,
+            0,
+            address(characterNFTs),
+            PitcherTokenID,
+            address(otherCharacterNFTs),
+            BatterTokenID
+        );
+        emit SessionResolved(
+            initialNumSessions + 2,
+            Outcome.HomeRun,
+            address(characterNFTs),
+            PitcherTokenID,
+            address(otherCharacterNFTs),
+            BatterTokenID
+        );
+        emit AtBatProgress(
+            initialNumAtBats + 1,
+            AtBatOutcome.HomeRun,
+            1,
+            0,
+            address(characterNFTs),
+            PitcherTokenID,
+            address(otherCharacterNFTs),
+            BatterTokenID
+        );
+        game.submitAtBat(pitcher, batter, pitches, swings, AtBatOutcome.HomeRun);
+
+        uint256 atBatID = game.NumAtBats();
+
+        assertEq(atBatID, initialNumAtBats + 1);
+        assertEq(game.NumSessions(), initialNumSessions + atBatLength);
+
+        AtBat memory atBat = game.getAtBat(atBatID);
+        assertEq(atBat.balls, 1);
+        assertEq(atBat.strikes, 0);
+        assertTrue(atBat.outcome == AtBatOutcome.HomeRun);
+
+        Session memory session;
+        for (uint256 i = 0; i < atBatLength; i++) {
+            session = game.getSession(game.AtBatSessions(atBatID, i));
+            if (i == 0) {
+                assertTrue(session.outcome == Outcome.Ball);
+            } else {
+                assertTrue(session.outcome == Outcome.HomeRun);
+            }
+        }
+    }
+
+    function test_submit_with_inconclusive_at_bat() public {
+        vm.prank(player1);
+        game.setTrustedExecutor(executor1, true);
+
+        vm.prank(player2);
+        game.setTrustedExecutor(executor1, true);
+
+        uint256 initialNumAtBats = game.NumAtBats();
+        uint256 initialNumSessions = game.NumSessions();
+
+        uint256 atBatLength = 6;
+
+        NFT memory pitcher = NFT({ nftAddress: address(characterNFTs), tokenID: PitcherTokenID });
+        NFT memory batter = NFT({ nftAddress: address(otherCharacterNFTs), tokenID: BatterTokenID });
+        Pitch[] memory pitches = new Pitch[](atBatLength);
+        Swing[] memory swings = new Swing[](atBatLength);
+
+        (pitches[0], swings[0]) = _generateStrike();
+        (pitches[1], swings[1]) = _generateBall();
+        (pitches[2], swings[2]) = _generateStrike();
+        (pitches[3], swings[3]) = _generateBall();
+        (pitches[4], swings[4]) = _generateFoul();
+        (pitches[5], swings[5]) = _generateBall();
+
+        vm.prank(executor1);
+        vm.expectRevert("Fullcount.submitAtBat: invalid at-bat - inconclusive");
+        game.submitAtBat(pitcher, batter, pitches, swings, AtBatOutcome.Strikeout);
```

Comment:
@kellan-simiotics : Maybe this test shouldn't have a conclusive proposed outcome. Can the proposed outcome be "in progress"?